src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php

```diff
@@ -50,7 +53,7 @@ public function create(string $resourceClass, string $property, array $options =
             $resourceClass = $childResourceClass;
         }
 
-        list($normalizationGroups, $denormalizationGroups) = $this->getEffectiveSerializerGroups($options, $resourceClass);
+        list($normalizationGroups, $denormalizationGroups) = $this->contextBuilder->createFromResourceClass($options, $resourceClass);
```

Comment:
We should deprecate the `$this->getEffectiveSerializerGroups` and propose to inject the contextbuilder instead